src/snowflake/snowpark/_internal/temp_table_auto_cleaner.py

```diff
@@ -33,74 +31,51 @@ def __init__(self, session: "Session") -> None:
                 f"Unexpected reference count {self.ref_count_map[name]} for table {name}"
             )
 
-    def process_cleanup(self) -> None:
-        while not self.stop_event.is_set():
-            try:
-                # it's non-blocking after timeout and become interruptable with stop_event
-                # it will raise an `Empty` exception if queue is empty after timeout,
-                # then we catch this exception and avoid breaking loop
-                table_name = self.queue.get(timeout=1)
-                self.drop_table(table_name)
-            except Empty:
-                continue
-
-    def drop_table(self, name: str) -> None:
+    def drop_table(self, name: str) -> None:  # pragma: no cover
         common_log_text = f"temp table {name} in session {self.session.session_id}"
-        logging.debug(f"Cleanup Thread: Ready to drop {common_log_text}")
+        logging.debug(f"Ready to drop {common_log_text}")
         try:
-            # TODO SNOW-1556553: Remove this workaround once multi-threading of Snowpark session is supported
-            with self.session._conn._conn.cursor() as cursor:
-                cursor.execute(
-                    f"drop table if exists {name} /* internal query to drop unused temp table */",
-                    _statement_params={DROP_TABLE_STATEMENT_PARAM_NAME: name},
-                )
-            logging.debug(f"Cleanup Thread: Successfully dropped {common_log_text}")
+            async_job = self.session.sql(
+                f"drop table if exists {name} /* internal query to drop unused temp table */",
+            )._internal_collect_with_tag_no_telemetry(
+                block=False, statement_params={DROP_TABLE_STATEMENT_PARAM_NAME: name}
+            )
+            logging.debug(
+                f"Dropping {common_log_text} with query id {async_job.query_id}"
+            )
         except Exception as ex:
             logging.warning(
-                f"Cleanup Thread: Failed to drop {common_log_text}, exception: {ex}"
+                f"Failed to drop {common_log_text}, exception: {ex}"
             )  # pragma: no cover
 
-    def is_alive(self) -> bool:
-        return self.cleanup_thread is not None and self.cleanup_thread.is_alive()
+    @property
+    def num_temp_tables_created(self) -> int:
+        return len(self.ref_count_map)
 
-    def start(self) -> None:
-        self.stop_event.clear()
-        if not self.is_alive():
-            self.cleanup_thread = Thread(target=self.process_cleanup)
-            self.cleanup_thread.start()
+    @property
+    def num_temp_tables_cleaned(self) -> int:
+        return sum(v == 0 for v in self.ref_count_map.values())
 
-    def stop(self) -> None:
-        """
-        The cleaner will stop immediately and leave unfinished temp tables in the queue.
-        """
-        self.stop_event.set()
-        if self.is_alive():
-            self.cleanup_thread.join()
+    def reset_reference_count_map(self) -> None:
```

Comment:
is that only for testing purpose? can you just access the class to clear the map for just testing purpose?

Reply:
removed